kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -350,82 +394,60 @@ class KinesisSink private (
-        .withMessageAttributes(
-          Map(
-            "kinesisKey" ->
-              new MessageAttributeValue()
-                .withDataType("String")
-                .withStringValue(eventWithId.event.key)
-          ).asJava
-        )
-        .withId(eventWithId.id.toString)
-      (eventWithId.id, batchReqEntry)
+  private val toSqsBatchEntry: Event => SendMessageBatchRequestEntry = { event =>
+    val b64EncodedMsg = encode(event.msg)
+    new SendMessageBatchRequestEntry(UUID.randomUUID.toString, b64EncodedMsg)
+      .withMessageAttributes(
+        Map(
+          "kinesisKey" ->
+            new MessageAttributeValue()
+              .withDataType("String")
+              .withStringValue(event.key)
+        ).asJava
+      )
   }
 
   private def createBatchRequest(queueUrl: String, batch: List[SendMessageBatchRequestEntry]) =
     new SendMessageBatchRequest()
       .withQueueUrl(queueUrl)
       .withEntries(batch.asJava)
 
-  private def sendToSqsWithRetry(
+  private def sendToSqs(
     sqs: SqsClientAndName,
-    batchEntryGroup: List[(UUID, SendMessageBatchRequestEntry)],
-    maxRetries: Int
-  ) = {
-    val current = TrieMap(batchEntryGroup: _*)
-    retry
-      .JitterBackoff(maxRetries, 1.second)
-      .apply { () =>
-        Future {
-          val batchRequest = createBatchRequest(sqs.sqsBufferName, current.values.toList)
-          val res = sqs.sqsClient.sendMessageBatch(batchRequest)
-          val failed = res.getFailed().asScala
-          if (failed.nonEmpty) {
-            val errors = failed.map(_.toString).mkString(", ")
-            log.error(s"Sending to SQS queue [${sqs.sqsBufferName}] failed with errors [$errors]")
-            val successIds =
-              res.getSuccessful().asScala.map(_.getId()).map(UUID.fromString).toList
-            log.info(
-              s"${successIds.size} out of ${current.size} from batch was successfully send to SQS queue: ${sqs.sqsBufferName}"
-            )
-            successIds.foreach(current.remove)
-            None // means: retry
-          } else {
-            log.info(
-              s"Batch of ${current.size} was successfully send to SQS queue: ${sqs.sqsBufferName}."
-            )
-            Some(res.getSuccessful.size) //means: do not retry (the number doesn't matter in this context)
-          }
-        }
-      }
-  }
+    batchEntryGroup: List[SendMessageBatchRequestEntry]
+  ) =
+    Future {
+      val batchRequest = createBatchRequest(sqs.sqsBufferName, batchEntryGroup)
+      val res = sqs.sqsClient.sendMessageBatch(batchRequest)
+      val failed = res.getFailed().asScala
+      if (failed.nonEmpty) {
+        // It could be improved even more by storing events (that failed to be sent to SQS) to some persistance storage
+        val errors = failed.map(_.toString).mkString(", ")
+        log.error(
+          s"Sending to SQS queue [${sqs.sqsBufferName}] failed with errors [$errors]. Dropping events."
+        )
+        log.info(
```

Comment:
Why don't we put this in the above log ?

Reply:
That's different logging levels. I thought the warning should be just warning. Could be merged if that is not the case, of course.